internal/provider/cluster_resource.go

```diff
@@ -94,6 +96,14 @@ func clusterResourceAttributes() map[string]schema.Attribute {
 			MarkdownDescription: "IP for the DNS service",
 			Required:            true,
 		},
+		"addon_kube_proxy": schema.BoolAttribute{
+			MarkdownDescription: "Enable KubeProxy Addon",
+			Optional:            true,
```

Comment:
you will also want `Computed: True`, as the foundry sets the value to its default value when it's unset. Otherwise, it will confuse the provider sdk when a value comes back after read.

In general, you can use the `install_disk_force_overwrite` boolean as an analogy: https://github.com/meltcloud/terraform-provider-meltcloud/blob/main/internal/provider/enrollment_image_resource.go#L92-L93

it follows the same pattern - the bool value is optionally set by the user, but after creation set by the foundry.

Reply:
for example, you also want to set `boolplanmodifier.RequiresReplace(),`.

maybe it also makes sense to clean it up and set the values in a separate function like this: https://github.com/meltcloud/terraform-provider-meltcloud/blob/main/internal/provider/enrollment_image_resource.go#L287-L300

(be aware that this only sets the `Computed` fields. The `Required` fields are only set during `Read`, not after `Create` (TF SDK doesn't like that)